modules/oe_content_featured_media_field/src/Plugin/Field/FieldWidget/FeaturedMediaAutocompleteWidget.php

```diff
@@ -0,0 +1,163 @@
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\EntityReferenceAutocompleteWidget;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Plugin implementation of the 'oe_featured_media_autocomplete' widget.
+ *
+ * @FieldWidget(
+ *   id = "oe_featured_media_autocomplete",
+ *   label = @Translation("Featured media"),
+ *   description = @Translation("An autocomplete entity reference field and a text field."),
+ *   field_types = {
+ *     "oe_featured_media"
+ *   }
+ * )
+ */
+class FeaturedMediaAutocompleteWidget extends EntityReferenceAutocompleteWidget implements ContainerFactoryPluginInterface {
+
+  /**
+   * Entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs FeaturedMediaAutocompleteWidget widget plugin.
+   *
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Field\FieldDefinitionInterface $field_definition
+   *   The definition of the field to which the widget is associated.
+   * @param array $settings
+   *   The widget settings.
+   * @param array $third_party_settings
+   *   Any third party settings.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   Entity type manager service.
+   */
+  public function __construct($plugin_id, $plugin_definition, FieldDefinitionInterface $field_definition, array $settings, array $third_party_settings, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($plugin_id, $plugin_definition, $field_definition, $settings, $third_party_settings);
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $plugin_id,
+      $plugin_definition,
+      $configuration['field_definition'],
+      $configuration['settings'],
+      $configuration['third_party_settings'],
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
+    $element['featured_media'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Featured media'),
+    ];
+
+    $element['featured_media'] += parent::formElement($items, $delta, $element, $form, $form_state);
+    // Unset title_display to ensure the element title is always visible.
```

Comment:
Why? The title display is configurable no? Why are we removing field API capability of a) choosing the title and b) showing/hiding it?

Reply:
I agree with you, I assigned the parent title to the fieldset title. But the the title_display can't be set on form_display config for example and we want to make sure that the Media item title of the reference field is always visible. The entity reference field is using this flag when value is unlimited to hide the title. If I don't usnet it, it will hide it. So the underlined title in the screen shot will disappear.
![image](https://user-images.githubusercontent.com/22004498/82788602-0b814f80-9e69-11ea-995a-d7e66ec3b102.png)